### PR TITLE
More docs

### PR DESCRIPTION
**Pull request by AdamGS**
More docs, trying to cleanup some of the main rustdoc page.

### CODE REVIEW

File: vortex-array/src/implementation.rs [hunk 1]
```diff
@@ -21,6 +21,10 @@ pub trait ArrayDef {
     type Encoding: ArrayEncoding + ArrayEncodingExt<D = Self>;
 }
 
+/// Macro to generate all the necessary code for a new type of array encoding. Including:
+/// 1. New Array type that implements `AsRef<Array>`, `GetArrayMetadata`, `ToArray`, `IntoArray`, and multiple useful `From`/`TryFrom` implementations.
+/// 1. New Encoding type that implements `ArrayEncoding`.
```
Comment (lwwmanning) [inline vortex-array/src/implementation.rs:26]: Numbering is 1, 1, 1

File: vortex-array/src/arrow/dtype.rs [hunk 2]
```diff
@@ -131,6 +131,7 @@ pub fn infer_schema(dtype: &DType) -> VortexResult<Schema> {
     Ok(builder.finish())
 }
 
+/// Try and convert a Vortex [`DType`] into an a Arrow [`DataType`]
```
Comment (lwwmanning) [inline vortex-array/src/arrow/dtype.rs]: "Try to"

File: vortex-array/src/lib.rs [hunk 3]
```diff
@@ -102,13 +108,15 @@ impl Array {
         }
     }
 
+    /// Returns an arrays with all of the array's child arrays.
```
Comment (lwwmanning) [inline vortex-array/src/lib.rs]: "A Vec of Arrays"

File: vortex-sampling-compressor/src/compressors/mod.rs [hunk 4]
```diff
@@ -81,7 +81,7 @@ pub trait EncoderMetadata {
 
 impl Display for CompressionTree<'_> {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
-        Display::fmt(self.compressor.id(), f)
+        write!(f, "{}", self.compressor.id())
```
Comment (robert3005) [inline vortex-sampling-compressor/src/compressors/mod.rs]: prefer the old variant

**Review by robert3005 — APPROVED** [review]

**Comment by lwwmanning** [comment]
Just a couple minor typos

File: vortex-array/src/implementation.rs (hunk 1 above)
Comment (gatesn) [reply vortex-array/src/implementation.rs:26]: Markdown auto-generates the right numbers

Comment (AdamGS) [reply vortex-array/src/implementation.rs:26]: \+ makes it easier to re-order/add more points (checked both VSCode and rustdoc renders it correctly)

File: vortex-array/src/arrow/dtype.rs [hunk 5]
```diff
@@ -131,6 +131,7 @@ pub fn infer_schema(dtype: &DType) -> VortexResult<Schema> {
     Ok(builder.finish())
 }
 
+/// Try to convert a Vortex [`DType`] into an a Arrow [`DataType`]
```
Comment (a10y) [inline vortex-array/src/arrow/dtype.rs:134]:
```suggestion
/// Try to convert a Vortex [`DType`] into an Arrow [`DataType`]
```

File: vortex-array/src/lib.rs [hunk 6]
```diff
@@ -62,9 +61,14 @@ pub mod flatbuffers {
     pub use vortex_flatbuffers::array::*;
 }
 
+/// A central type for all Vortex arrays, which are known length sequences of compressed data.
```
Comment (a10y) [inline vortex-array/src/lib.rs:64]:
```suggestion
/// A central type for all Vortex arrays, which are known length sequences of typed and possibly compressed data
```

**Pull request merged**